backend: Extract handleExternalProxy from createHeadlampHandler

hxrshxz opened the pull request:
## Summary
This PR refactors `createHeadlampHandler` by extracting the external proxy handler into a separate method, reducing function length and improving code maintainability.
## Related Issue
Fixes #3273 (partial)
## Changes
- Extracted `/externalproxy` handler into `handleExternalProxy` method (~108 lines)
- Added `oauthRequestMap` and `oauthMu` fields to `HeadlampConfig` struct
- Updated OIDC handlers to use struct fields instead of local variables
- Reduced `createHeadlampHandler` from ~530 to ~420 lines
## Steps to Test
1. Run the backend linter: `make backend-lint`
2. Run the backend tests: `make backend-test`
3. Verify the external proxy functionality works as expected
## Screenshots (if applicable)
N/A - Backend refactoring only
## Notes for the Reviewer
N/A

skoeva reviewed (commented):
could you adjust the commit message to match the PR title? take a look [here](https://headlamp.dev/docs/latest/contributing/#2-follow-commit-guidelines) for more info

illume reviewed (changes requested):
Seems there’s some tests failing.

To run them locally you can do:

> make backend-test

illume reviewed (changes requested):
Thanks for this.

If you want to continue this:
- Please see the open review comments?
- Probably handleExternalProxy should be moved into a separate module?

hxrshxz commented:
hey @illume! moved the external proxy handler to a seperate pkg/externalproxy module as you suggested. also fixed all the copilot review comments while i was at it (variable shadowing, removed redundant timeout, added graceful shutdown for oauth cleanup, etc). all tests passing now 👍

hxrshxz committed:
backend: externalproxy: Extract handleExternalProxy to seperate pkg/externalproxy module

illume commented:
Thanks for this! I will come back to try it out and look in more detail.

Could you please consider changing the git commit message to this?
> backend: externalproxy: Extract handleExternalProxy to seperate pkg/externalproxy module

illume reviewed (changes requested):
Looks like it's not compiling now.

illume reviewed (changes requested):
<!-- prch:feedback -->
Thanks for these changes.

<!-- prch:commit-guidelines -->
Can you please have a look at the git commits to see if they meet the contribution guidelines? We use a Linux kernel style of git commits. See the [contributing guide](https://headlamp.dev/docs/latest/development/contributing/#commit-guidelines) for general context, and please see previous git commits with git log for examples.

<details>
<summary>Commit guidelines</summary>

- Use atomic commits focused on a single change.
- Use the title format `<area>: <description of changes>`.
- Keep the title and body lines under 72 characters.
- Explain the intention and why the change is needed.
- Make commit titles meaningful and describe what changed.
- Do not add code that a later commit rewrites; squash or reorder commits instead.
- Do not include `Fixes #NN` in commit messages.

Good examples:

- `frontend: HomeButton: Fix so it navigates to home`
- `backend: config: Add enable-dynamic-clusters flag`

</details>

<!-- prch:copilot-comments -->
Can you please address the open review comments?

skoeva reviewed (commented):
thanks for the work here - sorry it sat so long.

closing this one - #3273 was closed as not planned, the branch no longer compiles and `headlamp.go` has moved a lot on `main` since. the review findings are being tracked separately. a narrower, behavior-preserving extraction would be welcome later
